.github/workflows/reusable-e2e.yml

```diff
@@ -61,15 +82,19 @@ jobs:
           chmod +x /home/runner/work/maci/.local/bin/circom
 
       - name: Generate zkeys
+        if: ${{ env.CHANGED == 'true' }}
```

Comment:
While using pre-generated artifacts speeds up e2e tests, it sidesteps CI's purpose of ensuring code and dependencies are always in a working state. Have we considered validating zkey files every time to catch potential issues early?

Reply:
if there's even 1 change in the circuits folder, the flow will be as before and just compile them from scratch so that tests run with the "new" circuits code